pkgs/by-name/ys/ysfx/package.nix

```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, fetchFromGitHub, cmake, pkg-config
+, freetype, juce, libX11, libXcursor, libXext, libXinerama, libXrandr, libglvnd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ysfx";
+  version = "2022-07-31";
+
+  src = fetchFromGitHub {
+    owner = "jpcima";
+    repo = "ysfx";
+    rev = "8077347ccf4115567aed81400281dca57acbb0cc";
+    hash = "sha256-pObuOb/PA9WkKB2FdMDCOd9TKmML+Sj2MybLP0YwT+8=";
+  };
+
+  # Provide latest dr_libs.
+  dr_libs = fetchFromGitHub {
+    owner = "mackron";
+    repo = "dr_libs";
+    rev = "e4a7765e598e9e54dc0f520b7e4416359bee80cc";
+    hash = "sha256-rWabyCP47vd+EfibBWy6iQY/nFN/OXPNhkuOTSboJaU=";
+  };
+
+  prePatch = ''
+    rmdir thirdparty/dr_libs
+    ln -s ${dr_libs} thirdparty/dr_libs
+  '';
```

Comment:
I see it's provided in `ysfx` repo as submodule, it's safe to use newer libs than main program here? Why not `fetchSubmodules = true` on `ysfx` repo?

Reply:
dr_libs increments its minor version numbers when it introduces API changes rather than bugfixes. ysfx fetches dr_libs with dr_flac 0.12.31, dr_mp3 0.6.31 and dr_wav 0.13.2, whereas the latest versions are 0.12.42, 0.6.38, and 0.13.15 respectively, so they should be compatible. dr_wav specifically has these changes since 0.13.2: https://github.com/mackron/dr_libs/blob/e4a7765e598e9e54dc0f520b7e4416359bee80cc/dr_wav.h#L8353-L8400 . The bugfixes seem important enough to warrant the update.